pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -320,3 +333,13 @@ func labelSelectors(datadogConfig config.Component) (namespaceSelector, objectSe
 
 	return namespaceSelector, objectSelector
 }
+
+// isOwnedByJob returns true if the pod is owned by a Job
+func isOwnedByJob(ownerReferences []metav1.OwnerReference) bool {
+	for _, owner := range ownerReferences {
+		if owner.APIVersion == "batch/v1" && owner.Kind == "Job" {
```

Comment:
to be futurproof, I would suggest to check only the apigroup name and not the version. 

something like
```suggestion
		if strings.HasPrefix(owner.APIVersion , "batch/") && owner.Kind == "Job" {
```

Reply:
Makes a lot of sense. Thanks for pointing this out